zetaclient/evm/evm_signer_test.go

```diff
@@ -0,0 +1,296 @@
+package evm
+
+import (
+	"path"
+	"testing"
+
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	corecommon "github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	appcontext "github.com/zeta-chain/zetacore/zetaclient/app_context"
+	"github.com/zeta-chain/zetacore/zetaclient/common"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+	corecontext "github.com/zeta-chain/zetacore/zetaclient/core_context"
+	"github.com/zeta-chain/zetacore/zetaclient/metrics"
+	"github.com/zeta-chain/zetacore/zetaclient/outtxprocessor"
+	"github.com/zeta-chain/zetacore/zetaclient/testutils"
+	"github.com/zeta-chain/zetacore/zetaclient/testutils/stub"
+)
+
+var (
+	// Dummy addresses as they are just used as transaction data to be signed
+	ConnectorAddress    = sample.EthAddress()
+	ERC20CustodyAddress = sample.EthAddress()
+)
+
+func getNewEvmSigner() (*Signer, error) {
+	mpiAddress := ConnectorAddress
+	erc20CustodyAddress := ERC20CustodyAddress
+	logger := common.ClientLogger{}
+	ts := &metrics.TelemetryServer{}
+	return NewEVMSigner(
+		corecommon.BscMainnetChain(),
+		stub.EVMRPCEnabled,
+		stub.NewTSSMainnet(),
+		config.GetConnectorABI(),
+		config.GetERC20CustodyABI(),
+		mpiAddress,
+		erc20CustodyAddress,
+		logger,
+		ts)
+}
+
+func getNewEvmChainClient() (*ChainClient, error) {
+	logger := common.ClientLogger{}
+	ts := &metrics.TelemetryServer{}
+	cfg := config.NewConfig()
+	tss := stub.NewTSSMainnet()
+
+	evmcfg := config.EVMConfig{Chain: corecommon.BscMainnetChain(), Endpoint: "http://localhost:8545"}
+	cfg.EVMChainConfigs[corecommon.BscMainnetChain().ChainId] = &evmcfg
+	coreCTX := corecontext.NewZetaCoreContext(cfg)
+	appCTX := appcontext.NewAppContext(coreCTX, cfg)
+
+	return NewEVMChainClient(appCTX, stub.NewZetaCoreBridge(), tss, "", logger, evmcfg, ts)
+}
+
+func getNewOutTxProcessor() *outtxprocessor.Processor {
+	logger := zerolog.Logger{}
+	return outtxprocessor.NewOutTxProcessorManager(logger)
+}
+
+func getCCTX() (*types.CrossChainTx, error) {
+	var cctx crosschaintypes.CrossChainTx
+	err := testutils.LoadObjectFromJSONFile(&cctx, path.Join("../", testutils.TestDataPathCctx, "cctx_56_68270.json"))
```

Comment:
no blocker. Just a FYI, I created a `LoadCctxByNonce` in testutils for future unit tests.